projects/core/src/cms/store/effects/page.effect.ts

```diff
@@ -93,20 +95,38 @@ export class PageEffects {
         Array.isArray(slot.components.component)
       ) {
         for (const component of slot.components.component) {
-          page.slots[slot.position].components.push({
-            uid: component.uid,
-            uuid: component.uuid,
-            catalogUuid: this.getCatalogUuid(component),
-            typeCode: component.typeCode,
-            flexType: component.flexType
-          });
+          page.slots[slot.position].components.push(
+            this.getComponentData(component)
+          );
         }
       }
     }
 
     return page;
   }
 
+  private getComponentData(
+    component: ContentSlotComponentData
+  ): ContentSlotComponentData {
+    const comp: ContentSlotComponentData = {
+      uid: component.uid,
+      uuid: component.uuid,
+      typeCode: component.typeCode
+    };
+
+    if (component.catalogUuid) {
+      comp.catalogUuid = component.catalogUuid;
```

Comment:
Shouldn't be `comp.catalogUuid = this.getCatalogUuid(component)` ?

Reply:
Also functionality changed a little, because before we always had `catalogUuid` on a component (sometimes probably undefined), just want to make sure it was intended.

Reply:
yup, that was on purpose. I think we should not pollute the HTML, only if we use it (which is when we're in SmartEdit). I've updated the unit test accordingly.

Reply:
I agree code can be a bit confusing. The `catalogUuid` is however available on the component and used from there before.

Reply:
you where right about the `catalogUuid`, thanks for being sharp!